images/bootstrap/runner.sh

```diff
@@ -41,94 +41,68 @@ setup_ca(){
-            cp /var/log/dockerd.log $ARTIFACTS
-        fi
-        echo "Cleaning up after docker"
-        docker ps -aq | xargs -r docker rm -f || true
-        kill "$(</var/run/docker.pid)" || true
-        wait "$(</var/run/docker.pid)" || true
+# runs custom podman data root cleanup binary and debugs remaining resources
+cleanup_pinc() {
+    if [[ "${PODMAN_IN_CONTAINER_ENABLED:-false}" == "true" ]]; then
+        echo "Cleaning up after podman"
+        podman ps -aq | xargs -r podman rm -f || true
+        kill "$(</var/run/podman.pid)" || true
+        wait "$(</var/run/podman.pid)" || true
     fi
 }
 
 early_exit_handler() {
-    cleanup_dind
+    cleanup_pinc
 }
 
 # setup certificates before anything gets started
 setup_ca
 
-# optionally enable ipv6 docker
-export DOCKER_IN_DOCKER_IPV6_ENABLED=${DOCKER_IN_DOCKER_IPV6_ENABLED:-false}
-if [[ "${DOCKER_IN_DOCKER_IPV6_ENABLED}" == "true" ]]; then
-    echo "Enabling IPV6 for Docker."
-    # configure the daemon with ipv6
-    mkdir -p /etc/docker/
-    cat <<EOF >/etc/docker/daemon.json
-{
-  "ipv6": true,
-  "fixed-cidr-v6": "fc00:db8:1::/64"
-}
-EOF
+# optionally enable ipv6
+export PODMAN_IN_CONTAINER_IPV6_ENABLED=${PODMAN_IN_CONTAINER_IPV6_ENABLED:-true}
+if [[ "${PODMAN_IN_CONTAINER_IPV6_ENABLED}" == "true" ]]; then
+    echo "Enabling IPV6."
     # enable ipv6
     sysctl net.ipv6.conf.all.disable_ipv6=0
     sysctl net.ipv6.conf.all.forwarding=1
     # enable ipv6 iptables
     modprobe -v ip6table_nat
 fi
 
-# Check if the job has opted-in to docker-in-docker availability.
-export DOCKER_IN_DOCKER_ENABLED=${DOCKER_IN_DOCKER_ENABLED:-false}
-if [[ "${DOCKER_IN_DOCKER_ENABLED}" == "true" ]]; then
-    echo "Docker in Docker enabled, initializing..."
-
-    export DOCKER_DEBUG=${DOCKER_DEBUG:-false}
-    if [[ "${DOCKER_DEBUG}" == "true" ]]; then
-        mkdir -p /etc/docker/
-        # TODO: do not rely on this file not existing!
-        cat <<EOF >/etc/docker/daemon.json
-{
-  "debug": true,
-  "log-level": "debug"
-}
-EOF
-    fi
-
-    printf '=%.0s' {1..80}; echo
-    # If we have opted in to docker in docker, start the docker daemon,
+export PODMAN_IN_CONTAINER_ENABLED=${PODMAN_IN_CONTAINER_ENABLED:-false}
+if [[ "${PODMAN_IN_CONTAINER_ENABLED}" == "true" ]]; then
+    echo "Podman in Container enabled, initializing in podman compatible mode..."
     (
-        if [ -f "/etc/default/docker" ]; then
-            source /etc/default/docker
-        fi
-        /usr/bin/dockerd \
-            -p /var/run/docker.pid \
-            --data-root=/docker-graph \
-            --init-path /usr/libexec/docker/docker-init \
-            --userland-proxy-path /usr/libexec/docker/docker-proxy \
-            ${DOCKER_OPTS} \
-                >/var/log/dockerd.log 2>&1 &
+        export HTTP_PROXY=${CONTAINER_HTTP_PROXY}
+        export HTTPS_PROXY=${CONTAINER_HTTPS_PROXY}
+	export KIND_EXPERIMENTAL_PROVIDER="podman"
+	mkdir -p ${XDG_RUNTIME_DIR}/podman
+        podman system service \
```

Comment:
If i run the new image
`docker run -it --privileged  -e PODMAN_IN_CONTAINER_ENABLED=true quay.io/kubevirtci/bootstrap:v20220816-91d8af2 bash`

I see `/podman/podman.sock` is created
because `XDG_RUNTIME_DIR` is empty here

btw it affects kubevirtci / kubevirt, because we look for `/run/podman/podman.sock`
wdyt about using `/run/podman/podman.sock` please ?
`XDG_RUNTIME_DIR` is empty anyhow
this way we won't need to export `KUBEVIRTCI_PODMAN_SOCKET`

Currently because the socket isn't found, it select docker / docker socket which are an alias to podman and its socket

Reply:
Yes I think we should update to the path that kubevirtci expects.

Reply:
Not tested yet https://github.com/kubevirt/project-infra/pull/2258
(lets see the presubmit job)